src/FileAdapter.js

```diff
@@ -1,156 +1,111 @@
+    ) {
+      return Promise.reject(
+        new Error(_.ERRORS.CANNOT_PROCESS(_.MESSAGES.WRONG_FILE_FORMAT))
+      )
     }
 
-    compile(args, options = {}) {
-        if (!this.isCompilable) {
-            // No need to compile
-            return Promise.resolve()
-        }
+    // assign the file type
+    this.detectType(fileStream)
 
-        return new Promise((resolve, reject) => {
-            try {
-                // Attempt to load the file 
-                const content = fs.readFileSync(this.path, 'utf8')
-
-                if (!content) {
-                    // Next let's make sure we stop right here for empty files
-                    resolve("")
-                    return
-                }
-
-                // Try to parse the file and catch syntax errors
-                const template = ejs.compile(content, {})
-
-                // Finally, let's see if we can validate it
-                const output = template(args)
-
-                // We're good
-                resolve(options.json ? JSON.parse(output, null, 2) : output)
-            } catch (error) {
-                reject(new Error(_.ERRORS.CANNOT_LOAD(error.message)))
-            }
-        })
+    if (!this.isCompilable) {
+      // specific type not implemented yet
+      return Promise.reject(
+        new Error(_.ERRORS.CANNOT_PROCESS(_.MESSAGES.NO_COMPATIBLE_TYPE))
+      )
     }
 
-    load(args, options = {}) {
-        if (!this.exists) {
-            // First make sure the file exists
-            return Promise.reject(new Error(_.ERRORS.CANNOT_LOAD('it does not exist')))
+    return new Promise((resolve, reject) => {
+      try {
+        if (this.type === 'IMAGE') {
+          const imageAdapter = new ImageAdapter()
+          resolve(imageAdapter.process(fileStream))
         }
-
-        // Let's see if this is a recognized file y]type 
-        this.detectType()
-
-        // Compile the file if necessary
-        return this.compile(args, options)
-    }
-
-    copy(dest) {
-        return new Promise((resolve, reject) => {
-            // Let's move the file over
-            fs.copySync(this.path, path.resolve(dest, this.filepath))
-            
-            resolve()
-        })    
-    }
-
-    save(dest, args = {}) {
-        if (!this.exists) {
-            // First make sure the file exists
-            return Promise.reject(new Error(_.ERRORS.CANNOT_SAVE('it does not exist')))
-        }
-
-        if (!fs.existsSync(dest)) {
-            // First make sure the destination location
-            return Promise.reject(new Error(_.ERRORS.CANNOT_SAVE('the destination does not exist')))
-        }
-
-        // Let's see if this is a recognized file type 
-        this.detectType()
-
-        // Create sub directories if necessary
-        const dir = path.resolve(this.dir, path.dirname(this.filepath))
-        fs.exists(dir) || fs.mkdirs(dir)
-
-        if (!this.isCompilable) {
```

Comment:
Move this into a base adapter. So the file adapter should just give you the stream, and then a CompilableAdapter or something should be a "base" adapter for adapters like JSONAdapter, JSAdapter, MarkdownAdapter, that will extend that and you could have the compile method in the base one.